examples/language-modeling/run_clm.py

```diff
@@ -0,0 +1,323 @@
+# coding=utf-8
+# Copyright 2018 The Google AI Language Team Authors and The HuggingFace Inc. team.
+# Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Fine-tuning the library models for causal language modeling (GPT, GPT-2, CTRL) on a text file or a dataset.
+"""
```

Comment:
From experience, users will understand that only GPT, GPT-2 and CTRL are supported by that script. I would put `(GPT, GPT-2, CTRL, ...)` instead, and provide a link:

```suggestion
"""
Fine-tuning the library models for causal language modeling (GPT, GPT-2, CTRL, ...) on a text file or a dataset.
Find the full list of model architectures that can be fine-tuned by this script on the documentation: 
https://huggingface.co/transformers/model_doc/auto.html#transformers.AutoModelWithLMHead
"""
```
But that might be a bit too much. Maybe adding a README would be simpler.

Reply:
Not AutomodelWithLMHead, just CausalLM, but I can add that.

Reply:
Maybe the link is overkill, I just had an issue with `(GPT, GPT-2, CTRL)` which seems to imply that only those three models are supported.

Reply:
Or this link? https://huggingface.co/models?filter=lm-head

Reply:
This works too but this shows checkpoints, whereas this script can also train from scratch so showing architectures would probably be better

Reply:
No this links shows all kinds of LM. The script will only work with a model that can be loaded with `AutoModelForCausalLM` (since it uses that class).

Reply:
Then https://huggingface.co/models?filter=causal-lm ?

Reply:
(the other one is the deprecated one, will remove soon)

Reply:
Great!